### PR TITLE
Ore Detectors

### PR DESCRIPTION
**Pull request by Geevies**
* Replaced ore radars with ore detectors. It display pings in the darkness when it spots the ore, even through walls.
* Mining stationbounds and drones now get ore detectors.
* You can now see psi-pings through walls again.

### CODE REVIEW

File: code/modules/mining/ore_detector.dm [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+/obj/item/ore_detector/Destroy()
+	deactivate()
+	return ..()
+
+/obj/item/ore_detector/update_icon()
+	icon_state = "ore_scanner[active ? "-active" : ""]"
+
+/obj/item/ore_detector/attack_self(mob/user)
+	if(!search_ore)
+		to_chat(user, SPAN_WARNING("You haven't set an ore to search for yet!"))
```
Comment (JohnWildkins) [inline code/modules/mining/ore_detector.dm:32]: QoL thing: maybe this should instead start the ore selection process?

**Review by JohnWildkins — APPROVED** [review]

File: code/modules/mining/ore_detector.dm (hunk 1 above)
Comment (Geevies) [reply code/modules/mining/ore_detector.dm:31]: i would consider it, but i just revamped it with a vueui so you can choose to detect multiple ore types, which would make it seem weird

**Comment by Geevies** [comment]
!review

File: code/modules/mining/ore_detector.dm [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+	name = "ore detector"
+	desc = "A device capable of locating and displaying ores to the average untrained hole explorer."
+	icon = 'icons/obj/contained_items/tools/ore_scanner.dmi'
+	icon_state = "ore_scanner"
+	item_state = "ore_scanner"
+	w_class = ITEMSIZE_SMALL
+	slot_flags = SLOT_BELT
+	force = 1
+	var/active = FALSE
+	var/mob/our_user
```
Comment (NonQueueingMatt) [inline code/modules/mining/ore_detector.dm:11]: Use a weakref. You also don't clean this up in Destroy.

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

File: code/modules/mining/ore_detector.dm (hunk 2 above)
Comment (Geevies) [reply code/modules/mining/ore_detector.dm]: did so, it's cleaned up in deactivate which is called in destroy

**Review by fernerr — APPROVED** [review]

**Pull request merged**